cmd/k8s-operator/operator.go

```diff
@@ -840,6 +853,48 @@ func proxyClassHandlerForConnector(cl client.Client, logger *zap.SugaredLogger)
 	}
 }
 
+// nodeHandlerForProxyGroup returns a handler that, for a given Node,
+// returns a list of reconcile requests for ProxyGroups that are referenced in a ProxyGroup
+// via the ProxyClass set for that ProxyGroup (.spec.proxyClass) containing a .spec.
+func nodeHandlerForProxyGroup(cl client.Client, logger *zap.SugaredLogger) handler.MapFunc {
+	return func(ctx context.Context, o client.Object) []reconcile.Request {
+		labels := o.GetLabels()
+		if len(labels) < 1 {
+			return nil
+		}
+
+		pgList := new(tsapi.ProxyGroupList)
+		if err := cl.List(ctx, pgList); err != nil {
+			logger.Debugf("error listing ProxyGroups for ProxyClass: %v", err)
+			return nil
+		}
+
+		// NOTE: (ChaosInTheCRD), we don't want to create multiple reconcile requests for the same
+		// ProxyGroup
+		reqs := make([]reconcile.Request, 0)
+		for _, pg := range pgList.Items {
+			if pg.Spec.ProxyClass != "" {
```

Comment:
We can also have default ProxyClass

Reply:
I've updated this to make sure that it checks for default `ProxyClass`.